sklearn/pipeline.py

```diff
@@ -609,12 +629,26 @@ def _transform_one(transformer, X, y, weight, **fit_params):
     return res * weight
 
 
-def _fit_transform_one(transformer, X, y, weight, **fit_params):
-    if hasattr(transformer, 'fit_transform'):
-        res = transformer.fit_transform(X, y, **fit_params)
-    else:
-        res = transformer.fit(X, y, **fit_params).transform(X)
-    # if we have a weight for this transformer, multiply output
+def _fit_transform_one(is_transform, clsname, message, transformer, weight, X,
+                       y, **fit_params):
```

Comment:
New parameters should be added at the end and probably have some reasonable defaults,
```py
def _fit_transform_one(transformer, weight, X, y,
                       is_transform=None, clsname=None, message=None, **fit_params):
   # possibly.
   # is there a case where the below is not the case?
   if clsname is None:
      clsname = transformer.__class__.__name__
   if is_transform is None:
      is_transform = isinstance(transformer, TransformerMixin)
```

Reply:
> The log message is expected to output that ColumnTransformer is the "parent" transformer.

OK, but then it should not be called `clsname` but `parent_clsname` or `message_clsname`. 

My points that overall this PR is good, however these logging changes do make the code logic harder to follow in some places, some of it might be unavoidable but  I am wondering aloud if that impact could be reduced.

Take `_fit_transform_one`. Granted it's a private function, but I expect it to roughly be usable as `_fit_transform_one(estimator, X, y, **kwargs)` (as it was before) following the standard scikit-learn API. Now we add a parameter `is_transformer` to it, which makes one wonder because usually if we have the transformer instance we know if it's a transformer or not. If it `is_transformer` does not correspond to the same estimator, it should be made explicit in the name. Actually it seems to indicate whether to do the transform or not, independently on the estimator nature which is confusing.

Similarly if the function is called `_fit_tranform_one` but then the transform is actually done only if the optional `is_transform` set to True with a `partial` function somewhere else in the code that makes the whole thing hard to follow. If we only do only a `fit` without a transform I would have expected that to happen somewhere inside a function called `fit`, with some additional code for logging.

Though I have not looked in this code in details, only at your changes, so feel to tell me if I am wrong :)